docs/advanced_topics/adding_reports.rst

```diff
@@ -0,0 +1,126 @@
+Adding reports
+==============
+
+Reports are views with listings of pages matching a specific query. They can also export these listings in spreadsheet format.
+They are found in the `Reports` submenu: by default, the `Locked Pages` report is provided, allowing an overview of locked pages on the site.
+
+It is possible to create your own custom reports in the Wagtail admin. To do this, you will need to subclass
+``wagtail.admin.views.reports.ReportView``, which provides basic listing and spreadsheet export functionality:
+
+.. code-block:: python
+
+    # <project>/views.py
+
+    from wagtail.admin.views.report.ReportView
+
+
+    class CustomReportView(ReportView):
+        pass
+
+
+Defining your report
+~~~~~~~~~~~~~~~~~~~~~
+
+The most important attributes and methods to customise to define your report are:
```

Comment:
are there other less important attributes and methods which aren't referenced here?

Reply:
Yes - in addition to the standard inherited `get_context`, etc methods from `BaseListView` and `TemplateResponseMixin`, there are a number of other methods in `SpreadsheetExportMixin` which you could in principle override. I wouldn't expect this to be necessary in most cases, though.